src/goto-symex/goto_symex.cpp

```diff
@@ -739,6 +743,73 @@ bool goto_symext::constant_propagate_case_change(
   return true;
 }
 
+bool goto_symext::constant_propagate_replace(
+  statet &state,
+  symex_assignt &symex_assign,
+  const function_application_exprt &f_l1)
+{
+  const auto &f_type = to_mathematical_function_type(f_l1.function().type());
+  const auto &length_type = f_type.domain().at(0);
+  const auto &char_type = to_pointer_type(f_type.domain().at(1)).subtype();
+
+  const refined_string_exprt &s = to_string_expr(f_l1.arguments().at(2));
+  const auto s_data_opt = try_evaluate_constant_string(state, s.content());
+
+  if(!s_data_opt)
+    return false;
+
+  auto &new_data = f_l1.arguments().at(4);
+  auto &old_data = f_l1.arguments().at(3);
+
+  if(f_l1.source_location().is_nil())
+    return false;
+
+  irep_idt target_overload = f_l1.source_location().get_function();
+
+  // If it's a symbol we're using the replace(Character, Character) overload.
+  // There's another that takes CharSequence which isn't dealt with.
+  if(target_overload == "java::java.lang.String.replace:(CC)Ljava/lang/String;")
```

Comment:
:no_entry_sign: check the type of the arguments instead

Reply:
I don't think this should block, either one achieves what's needed

Reply:
that's java specific code in symex

Reply:
Doh, right, sure. Use the types, as Romain said.